front/lib/resources/vault_resource.ts

```diff
@@ -283,6 +283,17 @@ export class VaultResource extends BaseResource<VaultModel> {
     };
   }
 
+  canWrite(auth: Authenticator) {
+    if (this.isRegular()) {
+      return auth.canWrite([this.acl()]);
+    }
+    return auth.isBuilder() && auth.canWrite([this.acl()]);
```

Comment:
This does not cover the system vault well does it?

Reply:
but this could be used with the system api key right?

Reply:
I think these functions should have a switch on the vault kind with a default assertNever so that they are revisited as we add new kind of vaults, eg public vaults :)

Reply:
system api key has system group, and is builder - so yes it can be used
+1 to use a switch/case

Reply:
@tdraier already done, can you check if LGTY or not? 🙏🏻